packages/flutter_tools/lib/src/isolated/devtools_launcher.dart

```diff
@@ -15,47 +19,92 @@ import '../resident_runner.dart';
 /// a devtools dep in google3.
 class DevtoolsServerLauncher extends DevtoolsLauncher {
   DevtoolsServerLauncher({
+    @required ProcessManager processManager,
+    @required String pubExecutable,
     @required Logger logger,
-  }) : _logger = logger;
+  })  : _processManager = processManager,
+        _pubExecutable = pubExecutable,
+        _logger = logger;
 
+  final ProcessManager _processManager;
+  final String _pubExecutable;
   final Logger _logger;
 
-  io.HttpServer _devtoolsServer;
+  io.Process _devToolsProcess;
+  Uri _devToolsUri;
+
+  static final RegExp _serveDevToolsPattern =
+      RegExp(r'Serving DevTools at ((http|//)[a-zA-Z0-9:/=_\-\.\[\]]+)');
 
   @override
-  Future<void> launch(Uri observatoryAddress) async {
+  Future<void> launch(Uri vmServiceUri) async {
+    if (_devToolsProcess != null && _devToolsUri != null) {
+      // DevTools is already running - open a chrome window to _devToolsUri.
+      await Chrome.start([_devToolsUri.toString()]);
+      return;
+    }
```

Comment:
When DevTools is already running (for example if a user hits 'v' twice from CLI, or in the future case where we will launch DevTools in machine mode at app start), we should just open a Chrome window to the existing DevTools uri on `launch` instead of spinning up another instance of devtools.

@jonahwilliams @helin24 thoughts?

Reply:
We'll want a way to disable opening a Chrome window when this is invoked from the IDE, since we may want to call `serve` but then open the browser embedded in the IDE instead of in a new window (or is this already in place due to one of the conditions?)

Reply:
Good point. See latest 2 commits. I think this should add what we need for that. Anything else that is missing?

Reply:
Yes, that looks great! thanks